spec/index.md

```diff
@@ -2368,7 +2370,8 @@ This method lists the snapshots of the canister identified by `canister_id`. Cur
 ### IC method `delete_snapshot` {#ic-take_snapshot}
 
 This method deletes a specified snapshot of the canister. An error will be returned if the snapshot is not found. 
-A snapshot cannot be found if it was previously deleted, replaced by a new snapshot through a `take_snapshot` request, or if the canister itself has been deleted.
+
+A snapshot cannot be found if it was previously deleted, replaced by a new snapshot through a `take_snapshot` request, or if the canister itself has been deleted or depleted of cycles.
```

Comment:
What if the canister is uninstalled? Do the snapshots get deleted?

Reply:
No, the controllers can still use the snapshot to restore the state of the canister.

Reply:
This should also be reflected in the formal semantics - clearing `S.snapshots[A.canister_id]` if a canister is deleted or runs out of cycles.